### DISCUSSION TITLE
Import comments from older version.

### DISCUSSION THREAD

**Discussion by halon176** [umputun/remark42#1621] [Q&A] [answered]
Hi, I installed remark42 on my old server a long time ago. I had to urgently save some data and reset everything. I had a folder in /var/www/remark42 with a remark42.conf file and a /var folder inside which there was a /backup folder containing all the daily backups.

I don't remember where I had the executable, but I'm sure I didn't install it with Docker. On the new server, I tried to reproduce the setup. I downloaded the executable and this time I put it directly in /var/www/remark42, and running it with the secret and url parameters works fine (I had saved the nginx configuration).

At this point, I tried to restore the backup but I can't understand the error:

```
/var/www/remark42# ./remark42.linux-amd64 restore --secret=mysecret  --url=http://127.0.0.1:8080 
remark42 local-20230121T13:26:39
2023/04/06 19:19:53.515 [INFO]  restore userbackup-{{.SITE}}-{{.YYYYMMDD}}.gz, site remark
2023/04/06 19:19:53.515 [INFO]  import var/backup/userbackup-remark-20230406.gz (native), site remark
2023/04/06 19:19:53.515 [ERROR] failed with can't open import file var/backup/userbackup-remark-20230406.gz: import failed, can't open var/backup/userbackup-remark-20230406.gz: open var/backup/userbackup-remark-20230406.gz: no such file or directory
2023/04/06 19:19:53.515 [ERROR] failed with can't open import file var/backup/userbackup-remark-20230406.gz: import failed, can't open var/backup/userbackup-remark-20230406.gz: open var/backup/userbackup-remark-20230406.gz: no such file or directory
can't open import file var/backup/userbackup-remark-20230406.gz: import failed, can't open var/backup/userbackup-remark-20230406.gz: open var/backup/userbackup-remark-20230406.gz: no such file or directory

```

Somehow it is looking for a file named var/backup/userbackup-remark-20230406.gz while mine is var/backup/backup-halon.cc-20230330.gz.

I have the impression that the executable does not see the environment around it, so neither remark42.conf.

What can I do in this case? How do I specify to remark which is the root folder of the files it should use? Thanks a lot!
```
/var/www/remark42# tree -L 2
.
|-- remark42.conf
|-- remark42.linux-amd64
`-- var
    |-- avatars
    |-- backup
    |-- halon.cc.db
    |-- pictures
    |-- pictures.staging
    `-- remark.db
```

**Comment by halon176** [comment, marked as answer]
I fixed it by creating the system service `/etc/systemd/system/remark42.service` to which I passed the various references, like this: 

```
[Unit]
Description=remark42
After=network.target

[Service]
Type=simple
EnvironmentFile=/var/www/remark42/remark42.conf
WorkingDirectory=/var/www/remark42
ExecStart=/var/www/remark42/remark42.linux-amd64 server
Restart=always
RestartSec=10
SyslogIdentifier=remark42
User=www-data
Group=www-data

[Install]
WantedBy=multi-user.target

```
it is a solution found somewhere on web and of which I have not found any mention in the documentation. In fact, from the documentation there is no way to understand how to pass the parameters of the config files and the workdir to the executable

**Reply by paskal** [reply]
Thanks a lot for documenting your finding! I'll write it down to configuration.

Or as alternative you can submit a PR yourself, I would be delighted ti merge it.

**Reply by halon176** [reply]
Yes, I can do it. However, I'm not sure where it's best to include such information. Perhaps a section in the CLI parameters?

**Reply by paskal** [reply]
https://remark42.com/docs/getting-started/installation/ is the right place I think.

**Reply by paskal** [reply]
I've published the systemd instruction as in the installation documentation [here](https://remark42.com/docs/getting-started/installation/#installation-as-a-systemd-service), including the WorkingDirectory directive as you suggested.

**Reply by halon176** [reply]
Thank you so much! Forgive me, I had completely forgotten, but I'm glad you did it.
